tests/backup/backup_resiliency_test.go

```diff
@@ -1825,3 +1826,169 @@ var _ = Describe("{CancelAllRunningRestoreJobs}", func() {
+							finalBackupName := fmt.Sprintf("%s-%s-nodelay-%s", BackupNamePrefix, namespace, RandomString(5))
+							err = CreateBackupWithValidation(ctx, finalBackupName, SourceClusterName, bkpLocationName, backupLocationUID, appContextsToBackup, labelSelectors, BackupOrgID, srcClusterUid, "", "", "", "")
+							dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s] of namespace (scheduled Context) [%s]", backupName, namespace))
+							backupNamespaceMap[namespace] = finalBackupName
+							break
+						}
+					} else {
+						log.Infof(fmt.Sprintf("Backup [%s] succeded with a network delay of [%d]", backupName, i))
+						log.Infof(fmt.Sprintf("Deleting the delay of %dms  to the applications nodes", i))
+						err := InjectNetworkErrorInNodes(appNodesMap[namespace], "delay", "del", 0, i)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Removing delay of %dms from nodes", i))
+					}
+				}
+			}
+		})
+
+		Step("Taking restore of backup when network delay was removed", func() {
+			log.InfoD("Taking restore of backup when network delay was removed")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{backupNamespaceMap[namespace]})
+				restoreName := fmt.Sprintf("%s-%v", RestoreNamePrefix, backupNamespaceMap[namespace])
+				err = CreateRestoreWithValidation(ctx, restoreName, backupNamespaceMap[namespace], make(map[string]string), make(map[string]string), DestinationClusterName, BackupOrgID, appContextsToBackup)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restore [%s]", restoreName))
+			}
+		})
+	})
+	JustAfterEach(func() {
+		defer EndPxBackupTorpedoTest(scheduledAppContexts)
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		backupNames, err := GetAllBackupsAdmin()
+		dash.VerifySafely(err, nil, fmt.Sprintf("Fetching all backups for admin"))
+		for _, backupName := range backupNames {
+			wg.Add(1)
+			go func(backupName string) {
+				defer GinkgoRecover()
+				defer wg.Done()
+				backupUid, err := Inst().Backup.GetBackupUID(ctx, backupName, BackupOrgID)
+				_, err = DeleteBackup(backupName, backupUid, BackupOrgID, ctx)
+				dash.VerifySafely(err, nil, fmt.Sprintf("Delete the backup %s ", backupName))
+				err = DeleteBackupAndWait(backupName, ctx)
+				dash.VerifySafely(err, nil, fmt.Sprintf("waiting for backup [%s] deletion", backupName))
+			}(backupName)
+		}
+		wg.Wait()
+		restoreNames, err := GetAllRestoresAdmin()
+		for _, restoreName := range restoreNames {
+			err = DeleteRestore(restoreName, BackupOrgID, ctx)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting restore [%s]", restoreName))
+		}
+		log.Infof("Deleting the deployed applications")
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = true
+		DestroyApps(scheduledAppContexts, opts)
```

Comment:
we dont need this EndPxBackupTorpedoTest will clean up the namespace created by testcase